pkg/k8s/watchers/cilium_node.go

```diff
@@ -129,27 +129,21 @@ func (k *K8sWatcher) ciliumNodeInit(ciliumNPClient client.Clientset, asyncContro
 // from kube-apiserver.
 func (k *K8sWatcher) GetCiliumNode(ctx context.Context, nodeName string) (*cilium_v2.CiliumNode, error) {
 	var (
-		err                      error
-		nodeInterface            interface{}
-		exists, getFromAPIServer bool
+		err           error
+		nodeInterface interface{}
+		exists        bool
 	)
 	k.ciliumNodeStoreMU.RLock()
 	// k.ciliumNodeStore might not be set in all invocations of GetCiliumNode,
-	// for example, during Cilium initialization GetCiliumNode is called from
-	// WaitForNodeInformation, which happens before ciliumNodeStore,
-	// so we will fallback to perform an API request to kube-apiserver.
+	// for example, when the key-value store is connected and the local store might
+	// have stale versions of the CiliumNode resource.
 	if k.ciliumNodeStore == nil {
-		getFromAPIServer = true
+		return k.clientset.CiliumV2().CiliumNodes().Get(ctx, nodeName, v1.GetOptions{})
 	} else {
 		nodeInterface, exists, err = k.ciliumNodeStore.GetByKey(nodeName)
 	}
 	k.ciliumNodeStoreMU.RUnlock()
 
-	if !exists || getFromAPIServer {
```

Comment:
I'm a little bit confused here, the commit message says that this removes the fallback API call, but we still do a fallback on line: 141 (with the only difference being that we don't attempt fallback if it's not in the local cache store?).

Reply:
> (with the only difference being that we don't attempt fallback if it's not in the local cache store?)

That's right, mentioning "fallback" I was referring to that. IOW, with the new implementation, when `forceCiliumNodeFromApiServer` is false (IOW the kvstore is not connected) we check the local store and, if `nodeName` is not found, we immediately return an error without querying the api server too.